Pilot/MessageSender.py

```diff
@@ -164,15 +171,22 @@ def sendMessage(self, msg, flag):
     hostCertificate = self.params.get('HostCertificate')
     CACertificate = self.params.get('CACertificate')
 
-    logging.debug("sending message from the REST Sender")
-    try:
-      requests.post(url,  # pylint: disable=undefined-variable
-                    json=msg,
-                    cert=(hostCertificate, hostKey),
-                    verify=CACertificate)
-    except (requests.exceptions.RequestException, IOError) as e:  # pylint: disable=undefined-variable
-      logging.error(e)
-      return False
+    if isPython2_6():
+      try:
+        requests.post(url,  # pylint: disable=undefined-variable
+                      json=msg,
+                      cert=(hostCertificate, hostKey),
+                      verify=CACertificate)
+      except (requests.RequestException,IOError) as e:
+        logging.error(e)
+        return False
+    else:
+      try:
+        requests.post(url, json=msg, cert=(hostCertificate, hostKey),
+                      verify=CACertificate)
+      except (requests.exceptions.RequestException,IOError) as e:
+        logging.error(e)
+        return False
```

Comment:
You can leave without this `if..else` block if in `backportRequests` you define `RequestException` as 

```
class exceptions(object):
  class RequestException(Exception):
    """Exception for requests"""

    def __init__(self, url, message):
      super(RequestException, self).__init__(url, message)
      self.url = url
      self.message = message

    def __str__(self):
      return '\nProblem with connecting to this url:\n{0}\nProblem:\n{1}'.format(self.url, self.message)
```

Reply:
Ok. Nice solution. I will use it